homeassistant/components/upnp/device.py

```diff
@@ -59,7 +59,7 @@ async def async_create_device(
         upnp_device = await async_create_upnp_device(hass, ssdp_location)
 
         # Create profile wrapper.
-        igd_device = IgdDevice(upnp_device, None)
+        igd_device = IgdDevice(upnp_device, None)  # type: ignore[arg-type]
```

Comment:
`async-upnp-client` doesn't allow None for the second parameter.
https://github.com/StevenLooman/async_upnp_client/blob/6a5e4b430034e5ca1bdf82ac21a3558076837a59/async_upnp_client/profiles/profile.py#L113

Reply:
I didn't want to change the code, but is it a bug? or should the underlying library be adjusted to allow None?

Reply:
This is a bug, if the `IgdDevice` is ever subscribed to (using `async_subscribe_services()`) it will crash. I think the underlying library should be adjusted to allow `None`.

Reply:
I have created an issue in the upstream library: https://github.com/StevenLooman/async_upnp_client/issues/125

I think we should still merge this, as it improves the type checking overall, and we can remove the `# type: ignore[arg-type]` when the upstream library is adjusted.

Reply:
Ah, mypy was disabled before for upnp... It did trigger on dlna_dms with the same thing. I'll fix this, but I'd rather fix this in home assistant the same way it was fixed with dlna_dms. I'll look into this somewhere this week, probably the weekend.

Reply:
I've released [0.26.0](https://pypi.org/project/async-upnp-client/), which allows for a `None`-value. Unfortunately, I didn't give it a default value, but perhaps specifying it explicit is not a bad thing.